cpp/include/cudf/hashing/detail/hash_functions.cuh

```diff
@@ -0,0 +1,103 @@
+/*
```

Comment:
This file was moved from `cpp/include/cudf/detail/utilities/hash_functions.cuh` and also split up
The `MurmurHash3_32` was moved to `murmurh_hash32.cuh`
Some utilities were only used by md5 and were moved into `md5_hash.cu`